examples/mass_spring_3d.py

```diff
@@ -0,0 +1,123 @@
+import taichi as ti
+try:
+    import taichi_glsl as tl
+    import taichi_three as t3
+except ImportError:
+    print('This example needs the extension library Taichi THREE to work.'
+          'Please run `pip install --user taichi_three` to install it.')
+
+import numpy as np
+import math
+ti.init(ti.gpu)
+
+### Parameters
+
+N = 128
+NN = N, N
+W = 1
+L = W / N
+gravity = 0.5
+stiffness = 16
+damping = 2
+steps = 15
+dt = 5e-3
+
+### Physics
+
+x = ti.Vector(3, ti.f32, NN)
+v = ti.Vector(3, ti.f32, NN)
+b = ti.Vector(3, ti.f32, NN)
+F = ti.Vector(3, ti.f32, NN)
+
+
+@ti.kernel
+def init():
+    for i in ti.grouped(x):
+        x[i] = tl.vec((i + 0.5) * L - 0.5, 0.8).xzy
+
+
+links = [tl.vec(*_) for _ in [(-1, 0), (1, 0), (0, -1), (0, 1)]]
```

Comment:
The new gif still doesn't look quite right, so I took a look at your code. Maybe you want
```suggestion
links = [tl.vec(*_) for _ in [(-1, 0), (1, 0), (0, -1), (0, 1), (-1, -1), (1, -1), (-1, 1), (1, 1)]]
```

Reply:
(Not sure if you need to take care of the rest length of these diagonal springs.)

Reply:
![video](https://user-images.githubusercontent.com/20640597/86505395-93f2f880-bdf6-11ea-955b-b9120ba8f10c.gif)
Cool! Changing the link between grids changes the behavior significantly.